src/nvim/os/fs.c

```diff
@@ -192,13 +192,15 @@ int os_open(const char* path, int flags, int mode)
 
 /// Get stat information for a file.
 ///
-/// @return OK on success, FAIL if a failure occurred.
+/// @return `true` on success, `false otherwise.
 static bool os_stat(const char *name, uv_stat_t *statbuf)
   FUNC_ATTR_NONNULL_ALL
 {
   uv_fs_t request;
   int result = uv_fs_stat(uv_default_loop(), &request, name, NULL);
```

Comment:
This might be a dumb idea, but it may be worth trying is to use a private loop instead of the default loop (see `stream_set_blocking()` for example). Just to see if MSAN changes its opinion.